src/ReactiveSocket.h

```diff
@@ -106,25 +108,28 @@ class ReactiveSocket {
   void onDisconnected(ReactiveSocketCallback listener);
   void onClosed(ReactiveSocketCallback listener);
 
-  void metadataPush(std::unique_ptr<folly::IOBuf> metadata);
-
   void tryClientResume(
       const ResumeIdentificationToken& token,
-      std::unique_ptr<DuplexConnection> newConnection,
+      std::shared_ptr<FrameTransport> frameTransport,
       std::unique_ptr<ClientResumeStatusCallback> resumeCallback);
 
   bool tryResumeServer(
       std::shared_ptr<FrameTransport> frameTransport,
       ResumePosition position);
 
+  folly::Executor& executor() {
```

Comment:
then transitively the returned value will be const Executor& (because it is reference member variable) which won't allow you to call methods in the executor.